crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs

```diff
@@ -0,0 +1,203 @@
+use holo_hash::ActionHash;
+use holochain_cascade::CascadeSource;
+use holochain_zome_types::{
+    record::{Record, SignedActionHashed},
+    Action,
+};
+use std::collections::{HashMap, HashSet};
+
+/// A collection of validation dependencies for the current set of DHT ops requiring validation.
+pub struct ValidationDependencies {
+    /// The state of each dependency, keyed by its hash.
+    states: HashMap<ActionHash, ValidationDependencyState>,
+    /// Tracks which dependencies have been accessed during a search for dependencies. Anything which
+    /// isn't in this set is no longer needed for validation and can be dropped from [`states`].
+    retained_deps: HashSet<ActionHash>,
+}
+
+impl Default for ValidationDependencies {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ValidationDependencies {
+    pub fn new() -> Self {
+        Self {
+            states: HashMap::new(),
+            retained_deps: HashSet::new(),
+        }
+    }
+
+    /// Check whether a given dependency is currently held.
```

Comment:
Held to me in the sys validation means that we have the action available locally, as opposed to just knowing its hash exists. It says nothing about the integrated or validated status. So yes it could be the cache or limbo that it's come from.

Should the terminology change here given this?

Reply:
@ThetaSinner I think we should have a definite bit of terminology for that. We could agree to use "hold" for that. I just want it to be an actual technical term and not something casual. If so:

something is "held" by me I can get at it locally somehow (this means it's either in the cache or awaiting validation)
something is "integrated" by me when I've put it, and all of its dependencies through the full validation process

If we want to keep "held" as a casual nontechnical term then we just need something else to replace it.